kafka-commons/src/test/java/uk/gov/homeoffice/digital/sas/kafka/producer/KafkaProducerServiceTest.java

```diff
@@ -62,4 +81,80 @@ void sendMessage_actionOnResource_messageIsSentWithCorrectArguments(KafkaAction
     assertThat(messageArgument.getValue().getResource()).isEqualTo(profile);
     assertThat(messageArgument.getValue().getAction()).isEqualTo(action);
   }
+
+  @ParameterizedTest
+  @EnumSource(value = KafkaAction.class)
+  void sendMessage_actionOnResource_onFailureInterruptLogged(KafkaAction action)
+      throws ExecutionException, InterruptedException {
+    ListAppender<ILoggingEvent> listAppender = getLoggingEventListAppender();
+
+    List<ILoggingEvent> logList = listAppender.list;
+
+    responseFuture = mock(CompletableFuture.class);
+
+    when(kafkaTemplate.send(any(), any(), any()))
+        .thenReturn(responseFuture);
+    Mockito.doThrow(InterruptedException.class).when(responseFuture).get();
+
+    kafkaProducerService.sendMessage(PROFILE_ID, profile, action);
+    assertThat(responseFuture.isDone()).isFalse();
+    assertThat(String.format(
+        KAFKA_FAILED_MESSAGE,
+        PROFILE_ID, "callisto-profile-topic",action.toString().toLowerCase())).isEqualTo(logList.get(0).getMessage());
+  }
+
+  @ParameterizedTest
+  @EnumSource(value = KafkaAction.class)
+  void sendMessage_actionOnResource_onFailureMessageLogged(KafkaAction action)
+      throws ExecutionException, InterruptedException {
+    ListAppender<ILoggingEvent> listAppender = getLoggingEventListAppender();
+
+    List<ILoggingEvent> logList = listAppender.list;
+
+    responseFuture = mock(CompletableFuture.class);
+
+    when(kafkaTemplate.send(any(), any(), any()))
+        .thenReturn(responseFuture);
+    Mockito.doThrow(ExecutionException.class).when(responseFuture).get();
+
+    kafkaProducerService.sendMessage(PROFILE_ID, profile, action);
+    assertThat(responseFuture.isDone()).isFalse();
+    assertThat(String.format(
```

Comment:
Minor: it should be assertThat(actual).equals(expected) rather than the other way around